src/compiler/checker.ts

```diff
@@ -2782,8 +2782,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 return !isPropertyImmediatelyReferencedWithinDeclaration(declaration, usage, /*stopAtAnyPropertyDeclaration*/ false);
             }
             else if (isParameterPropertyDeclaration(declaration, declaration.parent)) {
-                // foo = this.bar is illegal in esnext+useDefineForClassFields when bar is a parameter property
-                return !(getEmitScriptTarget(compilerOptions) === ScriptTarget.ESNext && useDefineForClassFields
+                // foo = this.bar is illegal in es2022+useDefineForClassFields when bar is a parameter property
+                return !(getEmitScriptTarget(compilerOptions) >= ScriptTarget.ES2022 && useDefineForClassFields

```

Comment:
@sandersn Was this just a case of #36425 not being updated once 

Actually, I'm pretty sure that this should just be `useDefineForClassFields` and _not_ check the script target. It seems like `getUseDefineForClassFields` already has the check for the script target and may have been missed in #42663.

Honestly, I suspect that there are other uses of the `useDefineForClassFields` which get this wrong or are at least redundant. Skimming, these are at least redundant:

- `isBlockScopedNameDeclaredBeforeUse`
- `checkAndReportErrorForInvalidInitializer`
- `getFirstTransformableStaticClassElement`

And these might be wrong and would be fixed like in this PR:

- `checkConstructorDeclarationDiagnostics`

Reply:
There of course may be a nuance I'm totally missing here.

Reply:
I completely agree with you. The checks for the emit target are unnecessary and should be removed.

Reply:
> Maybe it would be a good idea to introduce a new variable `emitStandardClassFields` that combines ES2022+ and useDefineForClassFields. Actually, useDefineForClassFields is combined with the ES2022 check often enough that it's likely that `var emitStandardClassFields` should _replace_ `var useDefinedForClassFields`.

I'm confused; isn't that what the `useDefineForClassFields` variable is after #42663?


```
var useDefineForClassFields = getUseDefineForClassFields(compilerOptions);

// ...
export function getUseDefineForClassFields(compilerOptions: CompilerOptions): boolean {
    return compilerOptions.useDefineForClassFields === undefined ? getEmitScriptTarget(compilerOptions) >= ScriptTarget.ES2022 : compilerOptions.useDefineForClassFields;
}
```

Reply:
Almost, but when `{ useDefineForClassFields: true, target: "ES2017" }`, getUseDefineForClassFields is true but emitStandardClassFields would be false.

Reply:
(emitStandardClassFields would be `compilerOptions.useDefineForClassFields !== false && getEmitScriptTarget(compilerOptions) >= ScriptTarget.ES2022`)